mne/channels/channels.py

```diff
@@ -740,7 +740,23 @@ def pick_types(self, meg=None, eeg=False, stim=False, eog=False,
             ias=ias, syst=syst, seeg=seeg, dipole=dipole, gof=gof, bio=bio,
             ecog=ecog, fnirs=fnirs, include=include, exclude=exclude,
             selection=selection)
-        return self._pick_drop_channels(idx)
+
+        self._pick_drop_channels(idx)
+
+        # remove dropped channel types from reject and flat
+        if getattr(self, 'reject', None) is not None:
+            ch_types = list(self.reject.keys())
+            for ch_type in ch_types:
```

Comment:
```suggestion
            for ch_type in self.reject.keys():
```

Reply:
I tried it this way initially, but you end up getting a `RuntimeError: dictionary changed size during iteration` if any keys are actually removed. I could do `for ch_type in list(self.reject.keys())` if you think it should be more compact.

Reply:
`for ch_type in list(self.reject)` should suffice, no need for `.keys()`